lib/http_aws_sigv4.c

```diff
@@ -66,6 +66,89 @@ static void sha256_to_hex(char *dst, unsigned char *sha, size_t dst_l)
   }
 }
 
+/*
+ * Decide in an encoding-independent manner whether a character in an
+ * URL must be escaped. The same criterion must be used in strlen_uri()
+ * and strcpy_uri().
+ */
+static bool urlchar_needs_escaping(int c, bool in_query)
+{
+  return !((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') ||
+           (c >= '0' && c <= '9') || c == '-' || c == '.' || c == '_'
+           || c == '~' ||
+           (!in_query && c == '/') ||
+           (in_query && c == '&'));
+}
+
+/*
+ * strlen_uri() returns the length of the given URI
+ * this fucntion is a modify version of strlen_url in urlapi.c
```

Comment:
strlen_uri() returns the length of the given URI this fu**nc**tion is a modif**ied** version of strlen_url in urlapi.c